content/reference/routing.md

```diff
@@ -0,0 +1,406 @@
+- `'navigate'` - emitted when any of the above events is emitted.
+
+Most of the time you'll be wanting to use the `pushState` event. `replaceState`
+and `popState` are much less common, although you might need them from time to
+time.
+
+_note:_ Anchor tags, buttons, and input submissions have slightly different
+use cases. `<a>` tags are meant for static links to pages.
+`<button>` tags are meant for conditional logic that doesn't always link to
+another page. And `<input type="submit">` tags are meant to be used to submit
+forms.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+var app = choo()
+
+var app = choo()
+app.route('/', view)                 // 1.
+app.route('/other', other)           // 2.
+app.mount('body')
+
+function view () {                   // 3.
+  return html`
+    <body>
+      <a href="/">Navigate</a>
+    </body>
+  `
+}
+
+function other (state, emit) {       // 4.
+  return html`
+    <body>
+      <button onclick=${onclick}>
+        Go Back
+      </button>
+    </body>
+  `
+
+  function onClick () {
+    emit('popState')
+  }
+}
+```
+
+1. Create an initial view for `/`.
+2. Create a second view for `/other`.
+3. The initial view renders a single link to `/other`.
+4. The second view renders a button. When clicked, it sends you back to the
+   previous route.
+
+## Listening For Route Changes
+As we briefly mentioned in the previous section, we can listen to route
+changes using the `'navigate'` event.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+var app = choo()
+
+var app = choo()
+app.use((state, emitter) => {            // 1.
+  emitter.on('navigate', (route) => {    // 2.
+    console.log(`Navigated to ${route}`) // 3.
+  })
+})
+```
+
+1. Create a new store.
+2. Listen to a navigate event.
+3. Whenever navigate is emitted, we log out what the new route is we're
+   navigating to. Route is a string here.
+
+## Hash Routing
+Sometimes when you deploy a static app, you can't control the server
+part of it. Changing the route might mean the server interprets it
+differently, causing problems to occur when reloading the page.
+
+To work around this, Choo supports hash routing. Instead of writing a
+route as `foobar.com/bin/baz`, you can write it as `foobar.com#bin/baz`.
+
+```js
+var choo = require('choo')
+var app = choo()
+
+app.route('/', view)
+app.route('#hi', view)
+app.route('#hi/hello', view)
+
+function view (state, emit) {
+  return html`
+    <body>${state.route}</body>
+  `
+}
+```
+
+## Page Anchors
+Another use of hashes in urls is to map to anchors on the page. This is
+commonly used for headings in articles. So when a link is shared,
+they're navigated to the right heading in the page.
+
+Choo supports page anchors out of the box. It tries to match anchors on
+the page first. If no matching anchor is found, Choo will try to find a
+matching route in the router. If no matching route is found, the regular
+fallback behavior occurs.
```

Comment:
`fallback behavior` is not best left unexplained, maybe add `[...]fallback behavior occurs, such as navigating to a 404 route.`